conans/client/cmd/export.py

```diff
@@ -252,14 +252,13 @@ def _capture_scm_auto_fields(conanfile, conanfile_dir, package_layout, output, i
         origin = scm.get_qualified_remote_url(remove_credentials=True)
         local_src_path = scm.get_local_path_to_url(origin)
         return scm_data, local_src_path
-
     if scm_data.url == "auto":
         origin = scm.get_qualified_remote_url(remove_credentials=True)
-        if not origin:
-            raise ConanException("Repo origin cannot be deduced")
         if scm.is_local_repository():
             output.warn("Repo origin looks like a local path: %s" % origin)
         output.success("Repo origin deduced by 'auto': %s" % origin)
+        if origin is None:
+            output.warn("origin is None, upload' command will prevent uploading recipes with None values in these fields.")
```

Comment:
I think it would be better to keep ``scm_data.url = "auto"`` and not assign it ``None``. It is more symmetric to the ``scm.revision`` behavior.

Reply:
And if necessary to change output messages, check for the "auto" string.